src/gpuarray_buffer_cuda.c

```diff
@@ -758,7 +776,7 @@ static gpudata *cuda_alloc(gpucontext *c, size_t size, void *data, int flags) {
    * to a multiple of FRAG_SIZE.  This also ensures that if we split a
    * block, the next block starts properly aligned for any data type.
    */
-  if (!(ctx->flags & GA_CTX_DISABLE_ALLOCATION_CACHE)) {
+  if (ctx->max_cache_size != 0) {
```

Comment:
max_cache_size = SIZE_T_MAX (or anything bigger than the GPU memory), initial_cache_size = 0